src/routes/forms.go

```diff
@@ -0,0 +1,250 @@
+package routes
+
+import (
+	"net/http"
+
+	"github.com/Real-Dev-Squad/wisee-backend/src/dtos"
+	"github.com/Real-Dev-Squad/wisee-backend/src/models"
+	"github.com/gin-gonic/gin"
+	"github.com/uptrace/bun"
+)
+
+func FormRoutes(rg *gin.RouterGroup, db *bun.DB) {
+	forms := rg.Group("/forms")
+
+	forms.POST("", func(ctx *gin.Context) {
+		var requestBody dtos.CreateUpdateFormRequestDto
+		if err := ctx.ShouldBindJSON(&requestBody); err != nil {
+			errObj := dtos.ResponseDto{
+				Message: "invalid request",
+				Error: &dtos.ErrorResponse{
+					Message: "invalid request body",
+					Detail:  err.Error(),
+				},
+			}
+			ctx.JSON(http.StatusBadRequest, errObj)
+			return
+		}
+
+		var form = &models.Form{
+			Content:     requestBody.Content,
+			CreatedById: requestBody.PerformedById,
+			Status:      models.DRAFT,
+			OwnerId:     requestBody.PerformedById,
+		}
+
+		if _, err := db.NewInsert().Model(form).Exec(ctx); err != nil {
+			errObj := dtos.ResponseDto{
+				Message: "something went wrong",
+				Error: &dtos.ErrorResponse{
+					Message: "error creating form",
+					Detail:  err.Error(),
+				},
+			}
+			ctx.JSON(http.StatusBadRequest, errObj)
+			return
+		}
+
+		var FormMetaData = &models.FormMetaData{
+			FormId: form.Id,
+		}
+
+		if _, err := db.NewInsert().Model(FormMetaData).Exec(ctx); err != nil {
+			errObj := dtos.ResponseDto{
+				Message: "something went wrong",
+				Error: &dtos.ErrorResponse{
+					Message: "error creating form meta data",
+					Detail:  err.Error(),
+				},
+			}
+			ctx.JSON(http.StatusBadRequest, errObj)
+			return
+		}
+
+		var resData = dtos.CreateUpdateGetFormResponseDto{
+			Id:          form.Id,
+			Content:     form.Content,
+			OwnerId:     form.OwnerId,
+			CreatedById: form.CreatedById,
+			Status:      string(form.Status),
+			CreatedAt:   form.CreatedAt.String(),
+			UpdatedAt:   form.UpdatedAt.String(),
+		}
+
+		resObj := dtos.ResponseDto{
+			Message: "form created successfully",
+			Data:    resData,
+		}
+
+		ctx.JSON(http.StatusCreated, resObj)
+	})
+
+	forms.GET("", func(ctx *gin.Context) {
```

Comment:
Won't this be behind auth, and it will be user-specific right?

Reply:
Correct but we are yet to build our authentication layer. So for now this is general only.